src/averageAges.js

```diff
@@ -1,64 +1,84 @@
 'use strict';
 
 /**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
  * @param {object[]} people
  * @param {number} century - optional
  *
  * @return {number}
  */
+
+const MALE = 'm';
+const FEMALE = 'f';
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const filteredArr = century
+    ? people.filter((person) => {
+      const personDeathCentury = Math.ceil(person.died / 100);
+
+      return century === personDeathCentury && person.sex === MALE;
+    })
+    : people.filter((person) => person.sex === MALE);
+
+  const ageSum = filteredArr.reduce((acc, person) => {
+    return acc + (person.died - person.born);
+  }, 0);
+
+  return ageSum / filteredArr.length;
 }
 
 /**
- * Implement calculateWomenAverageAge function
- *
- * Function returns average age of women in array. If `withChildren` is
- * specified then function calculates average age only for women with children
- *
- * Hint: To check if a woman has children you should find someone who mention
- * her as mother.
  *
  * @param {object[]} people
  * @param {boolean} withChildren - optional
  *
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const mothers = people.map(person => person.mother);
+  const filteredArr = withChildren
+    ? people.filter((person) => {
+      return mothers.includes(person.name) && person.sex === FEMALE;
+    })
+    : people.filter((person) => person.sex === FEMALE);
+
+  const ageSum = filteredArr.reduce((acc, person) => {
+    return acc + (person.died - person.born);
+  }, 0);
```

Comment:
Similar calculations are made in the previous function, this can be transferred to a separate function and used as a helper.